src/System.Linq/tests/ConcatTests.cs

```diff
@@ -251,20 +251,34 @@ public void CountOfConcatIteratorShouldThrowExceptionOnIntegerOverflow()
             var supposedlyLargeCollection = new DelegateBasedCollection<int> { CountWorker = () => int.MaxValue };
             var tinyCollection = new DelegateBasedCollection<int> { CountWorker = () => 1 };
 
+            Action<Action> assertThrows = (testCode) =>
+            {
+                // The full .NET framework uses unsigned arithmetic summing up collection counts.
+                // See https://github.com/dotnet/corefx/pull/6131.
+                if (PlatformDetection.IsFullFramework)
+                {
+                    testCode();
+                }
+                else
+                {
+                    Assert.Throws<OverflowException>(testCode);
```

Comment:
Maybe I have a lot of memory on my PC? I know that the System.Linq tests take a *long* time. Maybe this OOMs on a more memory constrained device!

> System.Linq.Tests  Total: 2577, Errors: 0, Failed: 0, Skipped: 6, Time: 373.870s